ep0099/ep0099.go

```diff
@@ -99,34 +98,14 @@ func (d *Dev) reset() error {
 // Up to 4 HATs can be stacked and each one need a different address to
 // work.
 func isValidAddress(address uint16) error {
-	validAddresses := [...]uint16{0x10, 0x11, 0x12, 0x13}
-
-	for _, addr := range validAddresses {
-		if address == addr {
-			return nil
-		}
+	switch address {
+	case 0x10, 0x11, 0x12, 0x13:
+		return nil
+	default:
+		return errInvalidAddress
 	}
-
-	return errInvalidAddress
 }
 
-func (d *Dev) isValidChannel(channel uint8) error {
-	if _, exists := d.state[channel]; !exists {
-		return errInvalidChannel
-	}
-	return nil
-}
-
-// EP-0099 offers 4 channels per board
-func buildChannelsList() map[uint8]State {
-	// Using a map instead of list since indexes of channels are not zero-based
-	// values. That would cause loops to have to correct channel ids while
-	// looping through items or reading/setting values.
-	// With maps, keys correspond to actual channels on the board.
-	return map[uint8]State{
-		1: StateOff,
-		2: StateOff,
-		3: StateOff,
-		4: StateOff,
-	}
+func (d *Dev) isValidChannel(channel uint8) bool {
```

Comment:
Doesn't need to be a method.

Reply:
Unless you're seeing a big toll on performance, I'd say it makes sense to have it centralized in a method than replicate the validation in multiple places.

Reply:
Oh, I meant it can be a stand alone function, it doesn't need a pointer receiver. What I meant is that this can be rewritten as:

```
func isValidChannel(channel uint8) bool {
  return channel >= 1 && channel <= 4
}
```

It's not material, so I'll commit and this can be changed later if you want.

Reply:
oh, I get it now.
it really makes sense.